Remove the #error to avoid issues with legacy not dealing with this

Index: bfd/bfd-in.h
===================================================================
--- bfd/bfd-in.h.orig
+++ bfd/bfd-in.h
@@ -23,11 +23,6 @@
 #ifndef __BFD_H_SEEN__
 #define __BFD_H_SEEN__
 
-/* PR 14072: Ensure that config.h is included first.  */
-#if !defined PACKAGE && !defined PACKAGE_VERSION
-#error config.h must be included before this header
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
